tests/Mutator/Boolean/EqualIdenticalTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator\Boolean;
+
+use Infection\Tests\Mutator\AbstractMutatorTestCase;
+
+/**
+ * @internal
+ */
+final class EqualIdenticalTest extends AbstractMutatorTestCase
+{
+    /**
+     * @dataProvider provideMutationCases
+     */
+    public function test_mutator($input, $expected = null)
+    {
+        $this->doTest($input, $expected);
+    }
+
+    public function provideMutationCases(): \Generator
+    {
+        yield 'It mutates equal operator into identical operator' => [
+            <<<'PHP'
+<?php
+
+$a == $b;
```

Comment:
Exactly under which circumstances these mutations are expected to fly under the radar? 

* If `$a == $b` is false, so is `$a === $b` false too. 
* If `$a != $b` is true, so is `$a !== $b` is true.
* If `$a != $b` is false, so is `$a !== $b` false too.
* If `$a == $b` is true, so is `$a === $b` true too, in most circumstances. Moreover, you can't possibly write a test for this mutation.

These new mutations bring nothing to the table unless proven otherwise.

Reply:
This mutator is intended for those who use `===` at all times, and only use `==` when strictly needed. Which is also why its not a default mutator, as its not usefull for all our users, and may produce a bunch of false positives if you have a different code style for your project.

Reply:
Well, what kind of faulty *tests* do you think these mutators are supposed to find?

Reply:
Lets say your application runs some other cli programs, and needs to check it return types. Because some of them have their own adapters for php you use that to run these programs.

But as there is no standard for it, some return the int 0 on success, while others return the sting 0 on success.

So you'd have a check somewhere that looks like this:
```php
if ($cliAdapter->getExitCode() != 0) {
    throw new NotSuccessfulException();
}
```

Now, if your tests only check here with the int 0, changing the `!=` to `!==` won't be caught. Which cold be improved by also adding a check against the string 0.

Reply:
The reason it should not be in the default set is for those who use `==` and `!=` almost exclusively. For them this mutator will give them a ton of mutators that can't be caught. 

Which is the same reason why i propose to make the `===` => `==` mutator not part of default, as people who use `===` exclusively will get a ton of mutators that can't be caught.

Reply:
Where an `==` &rarr; `===` mutator (and another) has a distinct and very specific purpose to work against very well known test deficiencies, which I tried to demonstrate in #394, and even if these mutators may cause some grief at times, I don't see the grief only as a valid reason for removal. I myself found some quite embarrassing testing bugs with an `==` &rarr; `===` mutator in my own code, and that's the whole purpose of mutation testing, by the way.

On the other hand these counterpart mutators are really quite vague, which I'm not a big fan of tbh. If you, yourself, do not wish to sponsor them, I don't see why we should have them, like, at all.